test/dynamo/test_higher_order_ops.py

```diff
@@ -0,0 +1,351 @@
+# Owner(s): ["module: dynamo"]
+import re
+import unittest
+
+import torch
+
+import torch._dynamo.test_case
+from torch._dynamo.utils import counters
+from torch._ops import wrap
+
+
+class MockBackend:
+    def __init__(self):
+        self.graphs = []
+
+    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+        self.graphs.append(gm)
+        return gm.forward
+
+
+global_var = torch.randn(3)
+global_num = 3.14
+
+
+class TestHigherOrderOps(torch._dynamo.test_case.TestCase):
+    def test_no_freevars(self):
+        mock = MockBackend()
+
+        def f(x):
+            return wrap(lambda x: torch.sin(x), x)
+
+        x = torch.randn(3)
+        expected = f(x)
+        result = torch.compile(f, backend=mock)(x)
+
+        self.assertEqual(result, expected)
+        self.assertEqual(len(mock.graphs), 1)
+        self.assertIsNotNone(re.search(r"wrap\(\w+, \w+\);", mock.graphs[0].code))
+
+    def test_capture_untracked_global(self):
+        counters.clear()
+        mock = MockBackend()
+
+        def f(x):
+            return wrap(lambda x: x + global_var, x)
+
+        x = torch.randn(3)
+        expected = f(x)
+        result = torch.compile(f, backend=mock)(x)
+
+        self.assertEqual(result, expected)
+        self.assertEqual(len(mock.graphs), 1)
+        # wrap(fn, x, global_var)
+        self.assertIsNotNone(re.search(r"wrap\(\w+, \w+, \w+\);", mock.graphs[0].code))
+        self.assertEqual(len(counters["graph_break"]), 0)
+
+    def test_capture_untracked_global_nested(self):
+        mock = MockBackend()
+        counters.clear()
+
+        @torch.compile(backend=mock)
+        def f(x):
+            return wrap(lambda x: wrap(lambda x: x + global_var, x), x)
+
+        x = torch.randn(3)
+        result = f(x)
+
+        self.assertEqual(result, x + global_var)
+        self.assertEqual(len(mock.graphs), 1)
+        gm = mock.graphs[0]
+        self.assertIsNotNone(re.search(r"wrap\(\w+, \w+, \w+\);", gm.code))
+        self.assertIsNotNone(re.search(r"wrap\(\w+, \w+, \w+\);", gm.cond_body_1.code))
```

Comment:
Just putting a note here in case anybody else is wondering why it's called `cond_xxx`. It appears back then "cond()" was the only user of `add_subgraph` so the word ended up being hardcoded there.

Reply:
Yes, that's on the list to change at some point